servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/gc/AbstractRestGCTest.java

```diff
@@ -0,0 +1,427 @@
+    // expect this commit to be expired as it is before cutoff time.
+    fillExpectedContents(branch1, 1, expectedExpired);
+    // commit for "table_2" on branch2
+    CommitOutput table2 =
+        commitSingleOp(
+            prefix, branch2, branch2.getHash(), 42, "cid_2", "table_2", "file1", null, null);
+    // expect this commit to be expired as it is before cutoff time.
+    fillExpectedContents(branch2, 1, expectedExpired);
+
+    final Instant cutoffTime = Instant.now();
+
+    // commits for "table_1" on branch 1
+    table1 =
+        commitSingleOp(
+            prefix, branch1, table1.hash, 44, "cid_1", "table_1", "file3", table1.content, null);
+
+    // commits for "table_2" on branch2
+    table2 =
+        commitSingleOp(
+            prefix, branch2, table2.hash, 44, "cid_2", "table_2", "file3", table2.content, null);
+    // expect this commit to be expired as branch2 is dropped below and there is no other reference
+    // to table_2.
+    fillExpectedContents(branch2, 1, expectedExpired);
+
+    // drop ref branch2
+    deleteBranch(branch2.getName(), table2.hash);
+
+    performGc(
+        cutoffTime,
+        null,
+        expectedExpired,
+        new HashSet<>(Arrays.asList(prefix + "_cid_1", prefix + "_cid_2")),
+        true);
+  }
+
+  @Test
+  public void testMultiRefSharedTable() throws BaseNessieClientServerException {
+    String prefix = "multiRefSharedTable";
+    Map<String, Set<Content>> expectedExpired = new HashMap<>();
+
+    Branch branch1 = createBranch(prefix + "_1");
+    // commit for "table_1" on branch1
+    CommitOutput b1table1 =
+        commitSingleOp(
+            prefix, branch1, branch1.getHash(), 42, "cid_1", "table_1", "file1", null, null);
+    // expect this commit to be expired as it is before cutoff time.
+    fillExpectedContents(branch1, 1, expectedExpired);
+    Branch branch2 = createBranch(prefix + "_2", Branch.of(branch1.getName(), b1table1.hash));
+    Branch branch3 = createBranch(prefix + "_3", Branch.of(branch1.getName(), b1table1.hash));
+    // commit for "table_1" on branch2
+    CommitOutput b2table1 =
+        commitSingleOp(
+            prefix,
+            branch2,
+            branch2.getHash(),
+            43,
+            "cid_1",
+            "table_1",
+            "file2",
+            b1table1.content,
+            null);
+    // expect this commit to be expired as it is before cutoff time.
+    fillExpectedContents(branch2, 1, expectedExpired);
+
+    final Instant cutoffTime = Instant.now();
+
+    // commit for "table_1" on branch1
+    b1table1 =
+        commitSingleOp(
+            prefix,
+            branch1,
+            b1table1.hash,
+            44,
+            "cid_1",
+            "table_1",
+            "file3",
+            b2table1.content,
+            null);
+    // expect this commit (after cutoff timestamp) to be expired as table is dropped below on
```

Comment:
I suspect, this is wrong.
The commit created here is after the cut-off time and should be considered "live" / usable via time-travel.
Same for line 305 below

Reply:
I got a comment for showing dropped table's content irrespective of commit time to be expired on my old PR from @rymurr. Hence I added this. This looks like a conflicting comment and we need to discuss and conclude on this

Reply:
we should support time travel for live tables only ?

Reply:
The table *is* live, because it is visible in a commit *after* cut-off time.

Reply:
but it is dropped after cut-off time. This table is alive in other reference. So, the commits on this table made after cut-off time in other reference is still kept. Only commits of this table on this reference is expired as this particular content is not present in any of the reference.